packages/SwingSet/docs/virtual-objects.md

```diff
@@ -15,118 +15,116 @@ A virtual object has a "kind", which defines what sort of behavior and state it
 ```
 
 Additional important details:
 
-- The set of state properties is determined by the `initialize` method.  The properties that exist when `initialize` returns _are_ the properties of the object.  State properties cannot thereafter be added or removed.
+- The set of state properties is determined by the `init` function.  The properties that exist when `init` returns _are_ the properties of the object.  State properties cannot thereafter be added or removed.
 
-- The values a state property may take are limited to things that are
-  serializable and which may be hardened (and, in fact, _are_ hardened and
-  serialized the moment they are assigned).  That is, you can replace what value
-  a state property _has_, but you cannot modify a state property's value in
-  place.  In other words, you can do things like:
+- The values a state property may take are limited to things that are serializable and which may be hardened (and, in fact, _are_ hardened and serialized the moment they are assigned).  That is, you can replace what value a state property _has_, but you cannot modify a state property's value in situ.  In other words, you can do things like:
 
-  ```state.zot = [1, 2, 3];```
+  `state.zot = [1, 2, 3];`
 
   but you can't do things like:
 
-  ```state.zot.push(4);```
+  `state.zot.push(4);`
 
 - A virtual object can be passed as a parameter in messages to other vats.  It will be passed by presence, just like any other non-data object you might send in a message parameter.
 
-- A virtual object's state may include references to other virtual objects.
+- A virtual object's state may include references to other virtual objects. The latter objects will be persisted separately and only deserialized as needed, so "swapping in" a virtual object that references other virtual objects does not entail swapping in the entire associated object graph.
 
-- The in-memory manifestations of a virtual object (which we call its "representatives") are not necessarily (or even usually) `===` to each other, even when they represent the same object, since a new representative is created for each act of deserialization.  They do, however, all operate on the same state, so changes to the state made by a method call to one representative will be visible to the other representatives.  A soon-to-arrive future version of the virtual object system will add a `sameKey` predicate that will allow you to compare two virtual object representatives to determine if they refer to the same virtual object instance.
+- The in-memory manifestations of a virtual object (which we call its "representatives") are not necessarily (or even usually) `===` to each other, even when they represent the same object, since a new representative is created for each act of deserialization.  They do, however, all operate on the same state, so changes to the state made by a method call to one representative will be visible to the other representatives.
```

Comment:
Please add a note about how we may tighten this so that JS identity does become canonical.